core/controllers/contributor_dashboard.py

```diff
@@ -328,10 +427,11 @@ class TranslatableTextHandler(base.BaseHandler):
     }
 
     @acl_decorators.open_access
-    def get(self):
+    def get(self) -> None:
         """Handles GET requests."""
-        language_code = self.normalized_request.get('language_code')
-        exp_id = self.normalized_request.get('exp_id')
+        assert self.normalized_request is not None
```

Comment:
Why is this the right thing to do, rather than check for the condition and raise an exception?

Reply:
Because at this point self.normalized_* can never be None:

As soon as we removed the class name from `handler_schema_constants.HANDLER_CLASS_NAMES_WITH_NO_SCHEMA`. The code executed further than this `if` clause 
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L386
and if the code is executed further, then self.normalized_* always populated. It may become empty dict but it will not stay as None.
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L504

Reply:
`asserts` can narrow down the types of variables only in those scopes where it is used. If we use assert in the base class then assert's changes will not be reflected in subclasses. Eg: [MYPY_PLAYGROUND](https://mypy-play.net/?mypy=latest&python=3.10&flags=strict%2Cdisallow-any-expr&gist=939b558999720f89cf7d661f8639a6e4)

And I think self.normalized_*  will never be a none value in the handlers that have schemas. If a handler is still not covered with schemas then we cannot use self.normalized_* there.

Reply:
OK, thanks for explaining. Yes, I think the main issue is that MyPy doesn't know that get() can only be called after dispatch(), so it regards the payload value as possibly None because that's what is defined in the init method.

Thanks for the explanation, I think we can leave this as-is.